wlutil/build.py

```diff
@@ -186,23 +186,78 @@ def buildWorkload(cfgName, cfgs, buildBin=True, buildImg=True):
     if ret != 0:
         raise RuntimeError("Error while building workload")
 
+def makeDrivers(boardDir, linuxSrc):
+    driverDirs = pathlib.Path(boardDir).glob("drivers/*")
+    makeCmd = "make -C " + str(linuxSrc) + " ARCH=riscv CROSS_COMPILE=riscv64-unknown-linux-gnu- "
+
+    # Prepare the linux source for building external drivers
+    run(["make", "ARCH=riscv", "CROSS_COMPILE=riscv64-unknown-linux-gnu-", "modules_prepare", jlevel], cwd=linuxSrc)
+
+    drivers = []
+    for driverDir in driverDirs:
+        run(makeCmd + "M=" + str(driverDir), cwd=driverDir, shell=True)
+        drivers.extend(list(driverDir.glob("*.ko")))
+
+    return drivers
+
+def setupBoardInitramfs(boardDir, linuxSrc):
+    log = logging.getLogger()
+
+    loadScript = initramfs_root / "loadDrivers.sh"
+
+    # clear out any leftover drivers from a previous build
+    for oldDriver in initramfs_root.glob("*.ko"):
+        oldDriver.unlink()
+    if loadScript.exists():
+        loadScript.unlink()
+
+    drivers = makeDrivers(boardDir, linuxSrc)
+
+    for driverPath in drivers:
+        shutil.copy(driverPath, initramfs_root)
+
+    with open(loadScript, 'w') as f:
+        for driverPath in drivers:
+            f.write("insmod /" + str(driverPath.name) + "\n")
```

Comment:
In theory, instead of doing this I could create a '/lib/modules/LINUX_KERNEL_VERSION/' directory and have the kernel auto-load the modules without this loadScript. Unfortunately, tracking down LINUX_KERNEL_VERSION is a pain and this loadScript works just fine. I guess the other way would be more idomatic for linux though. Not sure the right approach.